### PR TITLE
feat(chat-e2e): replay with message template

### PR DESCRIPTION
**Pull request by irinakartun**
**Description:**

replay with message template tests

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2856.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2856.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2856.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/tests/messageTemplate.test.ts [hunk 1]
```diff
@@ -744,19 +758,24 @@ dialTest(
...
     promptData,
     sendMessage,
     dataInjector,
     variableModalDialog,
+    variableModalAssertion,
     setTestIds,
   }) => {
-    setTestIds('EPMRTC-4298');
+    setTestIds('EPMRTC-4298', 'EPMRTC-4372');
     const aVar = 'A';
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/messageTemplate.test.ts:779]: Can we have overlaps with the variables names? Not familiar with the functionality, so I can be wrong. Probably it should not be hardcoded, but randomized.

File: apps/chat-e2e/src/tests/messageTemplate.test.ts [hunk 2]
```diff
@@ -846,7 +869,436 @@ dialTest(
...
+        await conversations.openEntityDropdownMenu(conversation.name);
+        await conversationDropdownMenu.selectMenuOption(MenuOptions.export);
+        const exportedData = await dialHomePage.downloadData(() =>
+          conversationDropdownMenu.selectMenuOption(
+            MenuOptions.withoutAttachments,
+          ),
+        );
+        await chatBar.deleteAllEntities();
+        await confirmationDialog.confirm({ triggeredHttpMethod: 'DELETE' });
+        await dialHomePage.importFile(exportedData, () =>
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/messageTemplate.test.ts:968]: don't forget to close the toast

File: apps/chat-e2e/src/tests/messageTemplate.test.ts [hunk 3]
```diff
@@ -846,7 +869,436 @@ dialTest(
...
+          .withConversationResource(conversation, PublishActions.ADD)
+          .build();
+        publication =
+          await publicationApiHelper.createPublishRequest(publishRequest);
+        publicationsToUnpublish.push(publication);
+      },
+    );
+
+    await dialTest.step(
+      'Create a replay conversation from "Approve required" section for publication request conversation',
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/messageTemplate.test.ts:1229]: I can barely imagine the reason for that functionality. Are we sure that it is needed?

Comment (irinakartun) [reply apps/chat-e2e/src/tests/messageTemplate.test.ts:1229]: Manual QA requested automation of this feature)

File: apps/chat-e2e/src/tests/messageTemplate.test.ts (hunk 1 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/messageTemplate.test.ts:779]: it doesn't make much sense to use random variable names for this functionality

File: apps/chat-e2e/src/tests/messageTemplate.test.ts (hunk 3 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/messageTemplate.test.ts:1229]: I believe the goal is to check whether shared/published conversations can also be parametrized

File: apps/chat-e2e/src/tests/messageTemplate.test.ts (hunk 2 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/messageTemplate.test.ts:968]: we have never closed the toast after import, does that make sense?

Comment (irinakartun) [reply apps/chat-e2e/src/tests/messageTemplate.test.ts:968]: fixed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2856.nightly-test.deltixhub.io
>E2E tests status: success

**Review by nartovm — APPROVED** [review]

**Pull request merged**